### PR TITLE
Add improved code intel support for Starlark

### PR DESCRIPTION
**Pull request by olafurpg**
First steps towards https://github.com/sourcegraph/sourcegraph/issues/37761 

Adds squirrel navigation for:
- local variables
- string literals in `load(...)` statements

Starlark has a more features that we need to implement eventually, but this is a decent milestone towards that goal.

## Test plan

See new Squirrel tests.

### CODE REVIEW

File: cmd/symbols/squirrel/lang_starlark.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	if exports != nil {
+		return &Node{
+			RepoCommitPath: node.RepoCommitPath,
+			Node:           exports,
+			Contents:       node.Contents,
+			LangSpec:       node.LangSpec,
+		}, nil
+	}
+	fmt.Println(node)
+	return nil, nil
```
Comment (Strum355) [inline cmd/symbols/squirrel/lang_starlark.go:49]:
```suggestion
	exports, err := starlarkBindingNamed(symbol, root)
	if err != nil || exports == nil {
		return nil, err
	}
	return &Node{
		RepoCommitPath: node.RepoCommitPath,
		Node:           exports,
		Contents:       node.Contents,
		LangSpec:       node.LangSpec,
	}, nil
```

I believe its functionally equivalent, and removes the `fmt.Println` :P

File: cmd/symbols/squirrel/lang_starlark.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+			return nil, err
+		}
+		exports, err := starlarkBindingNamed(symbol, *destinationRoot)
+		if err != nil {
+			return nil, err
+		}
+		return &Node{
+			RepoCommitPath: destinationRepoCommitPath,
+			Node:           exports,
+			//Node:           nil,
```
Comment (Strum355) [inline cmd/symbols/squirrel/lang_starlark.go:107]:
```suggestion
```

File: cmd/symbols/squirrel/lang_starlark.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	cursor.Exec(sitterQuery, getRoot(node.Node))
+
+	for {
+		match, ok := cursor.NextMatch()
+		if !ok {
+			return nil, nil
+		}
+
+		if len(match.Captures) < 3 {
+			panic(match.Captures)
```
Comment (Strum355) [inline cmd/symbols/squirrel/lang_starlark.go:69]: Number of panics here, can we replace them with breadcrumbs and/or returning errors? Im not exactly sure on the convention Chris has in place

**Review by Strum355 — APPROVED** [review]
Just some minor things to follow-up on whenever, otherwise :rocket:

File: cmd/symbols/squirrel/lang_starlark.go (hunk 3 above)
Comment (chrismwendt) [reply cmd/symbols/squirrel/lang_starlark.go]: You can return an error here pretty easily:

```suggestion
			return nil, errors.Newf("expected 3 captures in starlark query, got %d", len(match.Captures))
```

File: cmd/symbols/squirrel/lang_starlark.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+		if len(pathComponents) != 2 {
+			return nil, nil
+		}
+
+		directory := pathComponents[0]
+		filename := pathComponents[1]
+
+		if !strings.HasPrefix(directory, "//") {
+			panic(directory)
```
Comment (chrismwendt) [inline cmd/symbols/squirrel/lang_starlark.go]: You can return an error here, too:

```suggestion
			return nil, errors.Newf("expected starlark directory to be prefixed with \"//\", got %q", directory)
```

**Review by chrismwendt — APPROVED** [review]
🚀 I commandeered this as [suggested](https://sourcegraph.slack.com/archives/CHXHX7XAS/p1656679887724529) and pushed some commits.

**Pull request merged**